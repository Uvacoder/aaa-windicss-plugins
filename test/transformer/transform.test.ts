import { Processor } from 'windicss/lib';
import { transform } from 'windicss/helpers';

describe('transform', () => {
  it('tailwindcss-hero-patterns', () => {
    const processor = new Processor({
      plugins: [
        transform("tailwindcss-hero-patterns"),
      ]
    });
    expect(processor.interpret('heropattern-jigsaw-red-100 heropattern-overlappingcircles-blue-200').styleSheet.build()).toMatchSnapshot("css");
  })

  it('tailwindcss-truncate-multiline', () => {
    const processor = new Processor({
      theme: {
        truncate: {
          lines: {
              3: '3',
              5: '5',
              8: '8',
          }
        },
        extend: undefined
      },
      patterns: ["polka-dots", "signal"],
      plugins: [
        transform('tailwindcss-truncate-multiline')(),
      ]
    });
    expect(processor.interpret('truncate-3-lines truncate-5-lines').styleSheet.build()).toMatchSnapshot('css');
  })

  it('tailwindcss-fluid', () => {
    const processor = new Processor({
      plugins: [
        transform('tailwindcss-fluid')({
          textSizes: {
            sm: {
              min: '14px',
              max: '20px',
              minvw: '320px',
              maxvw: '1400px'
            }
          }
        })
      ]
    });
    expect(processor.interpret('text-sm-fluid').styleSheet.build()).toMatchSnapshot('css');
  });
})


it("tailwindcss-blend-mode", () => {
  const processor = new Processor({
    plugins: [
      transform('tailwindcss-blend-mode')(), // no options to configure
    ]
  });
  expect(processor.interpret('blend-normal blend-screen bg-blend-difference isolation-auto').styleSheet.build()).toMatchSnapshot('css');
})

it('tailwind-nord', () => {
  const processor = new Processor({
    plugins: [
      transform('tailwind-nord'),
    ]
  });
  expect(processor.interpret('bg-nord0 text-nord6').styleSheet.build()).toMatchSnapshot('css');
})

it('tailwind-color-vars', () => {
  const processor = new Processor({
    plugins: [
      transform('tailwind-color-vars')({
        colors: {
          'primary': 'rgba(0, 80, 200, 0.7)',
          'black': 'black',
        },
        // default value, this will give passed values priority
        strategy: 'override',  
      })
    ]
  });
  expect(processor.preflight(undefined, false, false, true).build()).toMatchSnapshot('css');
})

it('tailwindcss-triangle-after', () => {
  const colors = {
    blue: '#1c1c1e',
    'blue-darker': '#1c1c1c',
  };
  const processor = new Processor({
    plugins: [
      transform('tailwindcss-triangle-after')({
        triangles: {
          select: {
            color: colors['blue'],
            direction: 'down',
            size: [10, 6],
          },
          next: {
            color: colors['blue-darker'],
            direction: 'right',
            right: '2rem',
            top: '3rem',
            size: 12
          }
        },
      }),
    ]
  })
  expect(processor.interpret('triangle-after-select triangle-after-next').styleSheet.build()).toMatchSnapshot('css');
})

it('tailwindcss-gradients', () => {
  const processor = new Processor({
    theme: {
      colors: {
        'red': '#f00',
        'blue': '#00f',
      },
      linearGradientColors: theme => theme('colors'),
      radialGradientColors: theme => theme('colors'),
      conicGradientColors: theme => theme('colors'),
    },
    plugins: [
      transform('tailwindcss-gradients'),
    ]
  });
  expect(processor.interpret('bg-gradient-tl-red bg-radial-br-red bg-conic-tr-red').styleSheet.build()).toMatchSnapshot('css');
});

it('tailwindcss-border-gradients', () => {
  const processor = new Processor({
    theme: {
      linearBorderGradients: {
        colors: {
          'red': '#f00',
        },
      },
    },
    plugins: [
      transform('tailwindcss-border-gradients')(),
    ],
  });
  expect(processor.interpret('border-gradient-t-red border-gradient-tl-red').styleSheet.build()).toMatchSnapshot('css');
})

it('tailwindcss-elevation', () => {
  const processor = new Processor({
    plugins: [
      transform('tailwindcss-elevation')([]),
    ]
  });
  expect(processor.interpret('elevation-0 elevation-2 elevation-21').styleSheet.build()).toMatchSnapshot('css');
})