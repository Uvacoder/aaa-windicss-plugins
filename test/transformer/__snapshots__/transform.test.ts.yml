transform / tailwind-color-vars / css / 0: |-
  :root {
    --primary: rgba(0, 80, 200, 0.7);
    --black: black;
  }
transform / tailwind-nord / css / 0: |-
  .bg-nord0 {
    --tw-bg-opacity: 1;
    background-color: rgba(46, 52, 64, var(--tw-bg-opacity));
  }
  .text-nord6 {
    --tw-text-opacity: 1;
    color: rgba(236, 239, 244, var(--tw-text-opacity));
  }
transform / tailwindcss-blend-mode / css / 0: |-
  .blend-normal {
    mix-blend-mode: normal;
  }
  .blend-screen {
    mix-blend-mode: screen;
  }
  .bg-blend-difference {
    background-blend-mode: difference;
  }
  .isolation-auto {
    isolation: auto;
  }
transform / tailwindcss-border-gradients / css / 0: |-
  .border-gradient-t-red {
    border-image: linear-gradient(to top, rgba(255, 0, 0, 0), #f00) 1;
  }
  .border-gradient-tl-red {
    border-image: linear-gradient(to top left, rgba(255, 0, 0, 0), #f00) 1;
  }
transform / tailwindcss-fluid / css / 0: |-
  .text-sm-fluid {
    font-size: 14px;
  }
  @media (min-width: 320px) {
    .text-sm-fluid {
      font-size: calc(14px + 6 * (100vw - 320px) / 1080);
    }
  }
  @media (min-width: 1400px) {
    .text-sm-fluid {
      font-size: 20px;
    }
  }
transform / tailwindcss-gradients / css / 0: |-
  .bg-gradient-tl-red {
    background-image: linear-gradient(to top left, rgba(255, 0, 0, 0), #f00);
  }
  .bg-radial-br-red {
    background-image: radial-gradient(closest-side at bottom right, #f00, rgba(255, 0, 0, 0));
  }
  .bg-conic-tr-red {
    background-image: conic-gradient(at top right, #f00, rgba(255, 0, 0, 0));
  }
transform / tailwindcss-hero-patterns / css / 0: |-
  .heropattern-jigsaw-red-100 {
    background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='192' height='192' viewBox='0 0 192 192'%3E%3Cpath fill='%23fee2e2' fill-opacity='1' d='M192 15v2a11 11 0 0 0-11 11c0 1.94 1.16 4.75 2.53 6.11l2.36 2.36a6.93 6.93 0 0 1 1.22 7.56l-.43.84a8.08 8.08 0 0 1-6.66 4.13H145v35.02a6.1 6.1 0 0 0 3.03 4.87l.84.43c1.58.79 4 .4 5.24-.85l2.36-2.36a12.04 12.04 0 0 1 7.51-3.11 13 13 0 1 1 .02 26 12 12 0 0 1-7.53-3.11l-2.36-2.36a4.93 4.93 0 0 0-5.24-.85l-.84.43a6.1 6.1 0 0 0-3.03 4.87V143h35.02a8.08 8.08 0 0 1 6.66 4.13l.43.84a6.91 6.91 0 0 1-1.22 7.56l-2.36 2.36A10.06 10.06 0 0 0 181 164a11 11 0 0 0 11 11v2a13 13 0 0 1-13-13 12 12 0 0 1 3.11-7.53l2.36-2.36a4.93 4.93 0 0 0 .85-5.24l-.43-.84a6.1 6.1 0 0 0-4.87-3.03H145v35.02a8.08 8.08 0 0 1-4.13 6.66l-.84.43a6.91 6.91 0 0 1-7.56-1.22l-2.36-2.36A10.06 10.06 0 0 0 124 181a11 11 0 0 0-11 11h-2a13 13 0 0 1 13-13c2.47 0 5.79 1.37 7.53 3.11l2.36 2.36a4.94 4.94 0 0 0 5.24.85l.84-.43a6.1 6.1 0 0 0 3.03-4.87V145h-35.02a8.08 8.08 0 0 1-6.66-4.13l-.43-.84a6.91 6.91 0 0 1 1.22-7.56l2.36-2.36A10.06 10.06 0 0 0 107 124a11 11 0 0 0-22 0c0 1.94 1.16 4.75 2.53 6.11l2.36 2.36a6.93 6.93 0 0 1 1.22 7.56l-.43.84a8.08 8.08 0 0 1-6.66 4.13H49v35.02a6.1 6.1 0 0 0 3.03 4.87l.84.43c1.58.79 4 .4 5.24-.85l2.36-2.36a12.04 12.04 0 0 1 7.51-3.11A13 13 0 0 1 81 192h-2a11 11 0 0 0-11-11c-1.94 0-4.75 1.16-6.11 2.53l-2.36 2.36a6.93 6.93 0 0 1-7.56 1.22l-.84-.43a8.08 8.08 0 0 1-4.13-6.66V145H11.98a6.1 6.1 0 0 0-4.87 3.03l-.43.84c-.79 1.58-.4 4 .85 5.24l2.36 2.36a12.04 12.04 0 0 1 3.11 7.51A13 13 0 0 1 0 177v-2a11 11 0 0 0 11-11c0-1.94-1.16-4.75-2.53-6.11l-2.36-2.36a6.93 6.93 0 0 1-1.22-7.56l.43-.84a8.08 8.08 0 0 1 6.66-4.13H47v-35.02a6.1 6.1 0 0 0-3.03-4.87l-.84-.43c-1.59-.8-4-.4-5.24.85l-2.36 2.36A12 12 0 0 1 28 109a13 13 0 1 1 0-26c2.47 0 5.79 1.37 7.53 3.11l2.36 2.36a4.94 4.94 0 0 0 5.24.85l.84-.43A6.1 6.1 0 0 0 47 84.02V49H11.98a8.08 8.08 0 0 1-6.66-4.13l-.43-.84a6.91 6.91 0 0 1 1.22-7.56l2.36-2.36A10.06 10.06 0 0 0 11 28 11 11 0 0 0 0 17v-2a13 13 0 0 1 13 13c0 2.47-1.37 5.79-3.11 7.53l-2.36 2.36a4.94 4.94 0 0 0-.85 5.24l.43.84A6.1 6.1 0 0 0 11.98 47H47V11.98a8.08 8.08 0 0 1 4.13-6.66l.84-.43a6.91 6.91 0 0 1 7.56 1.22l2.36 2.36A10.06 10.06 0 0 0 68 11 11 11 0 0 0 79 0h2a13 13 0 0 1-13 13 12 12 0 0 1-7.53-3.11l-2.36-2.36a4.93 4.93 0 0 0-5.24-.85l-.84.43A6.1 6.1 0 0 0 49 11.98V47h35.02a8.08 8.08 0 0 1 6.66 4.13l.43.84a6.91 6.91 0 0 1-1.22 7.56l-2.36 2.36A10.06 10.06 0 0 0 85 68a11 11 0 0 0 22 0c0-1.94-1.16-4.75-2.53-6.11l-2.36-2.36a6.93 6.93 0 0 1-1.22-7.56l.43-.84a8.08 8.08 0 0 1 6.66-4.13H143V11.98a6.1 6.1 0 0 0-3.03-4.87l-.84-.43c-1.59-.8-4-.4-5.24.85l-2.36 2.36A12 12 0 0 1 124 13a13 13 0 0 1-13-13h2a11 11 0 0 0 11 11c1.94 0 4.75-1.16 6.11-2.53l2.36-2.36a6.93 6.93 0 0 1 7.56-1.22l.84.43a8.08 8.08 0 0 1 4.13 6.66V47h35.02a6.1 6.1 0 0 0 4.87-3.03l.43-.84c.8-1.59.4-4-.85-5.24l-2.36-2.36A12 12 0 0 1 179 28a13 13 0 0 1 13-13zM84.02 143a6.1 6.1 0 0 0 4.87-3.03l.43-.84c.8-1.59.4-4-.85-5.24l-2.36-2.36A12 12 0 0 1 83 124a13 13 0 1 1 26 0c0 2.47-1.37 5.79-3.11 7.53l-2.36 2.36a4.94 4.94 0 0 0-.85 5.24l.43.84a6.1 6.1 0 0 0 4.87 3.03H143v-35.02a8.08 8.08 0 0 1 4.13-6.66l.84-.43a6.91 6.91 0 0 1 7.56 1.22l2.36 2.36A10.06 10.06 0 0 0 164 107a11 11 0 0 0 0-22c-1.94 0-4.75 1.16-6.11 2.53l-2.36 2.36a6.93 6.93 0 0 1-7.56 1.22l-.84-.43a8.08 8.08 0 0 1-4.13-6.66V49h-35.02a6.1 6.1 0 0 0-4.87 3.03l-.43.84c-.79 1.58-.4 4 .85 5.24l2.36 2.36a12.04 12.04 0 0 1 3.11 7.51A13 13 0 1 1 83 68a12 12 0 0 1 3.11-7.53l2.36-2.36a4.93 4.93 0 0 0 .85-5.24l-.43-.84A6.1 6.1 0 0 0 84.02 49H49v35.02a8.08 8.08 0 0 1-4.13 6.66l-.84.43a6.91 6.91 0 0 1-7.56-1.22l-2.36-2.36A10.06 10.06 0 0 0 28 85a11 11 0 0 0 0 22c1.94 0 4.75-1.16 6.11-2.53l2.36-2.36a6.93 6.93 0 0 1 7.56-1.22l.84.43a8.08 8.08 0 0 1 4.13 6.66V143h35.02z'%3E%3C/path%3E%3C/svg%3E");
  }
  .heropattern-overlappingcircles-blue-200 {
    background-image: url("data:image/svg+xml,%3Csvg width='80' height='80' viewBox='0 0 80 80' xmlns='http://www.w3.org/2000/svg'%3E%3Cg fill='none' fill-rule='evenodd'%3E%3Cg fill='%23bfdbfe' fill-opacity='1'%3E%3Cpath d='M50 50c0-5.523 4.477-10 10-10s10 4.477 10 10-4.477 10-10 10c0 5.523-4.477 10-10 10s-10-4.477-10-10 4.477-10 10-10zM10 10c0-5.523 4.477-10 10-10s10 4.477 10 10-4.477 10-10 10c0 5.523-4.477 10-10 10S0 25.523 0 20s4.477-10 10-10zm10 8c4.418 0 8-3.582 8-8s-3.582-8-8-8-8 3.582-8 8 3.582 8 8 8zm40 40c4.418 0 8-3.582 8-8s-3.582-8-8-8-8 3.582-8 8 3.582 8 8 8z' /%3E%3C/g%3E%3C/g%3E%3C/svg%3E");
  }
transform / tailwindcss-triangle-after / css / 0: |-
  .triangle-after-select {
    position: relative;
  }
  .triangle-after-select::after {
    border-color: transparent;
    border-style: solid;
    content: "";
    position: absolute;
    pointer-events: none;
    top: 50%;
    transform: translateY(-50%);
    right: 1rem;
    border-top-color: #1c1c1e;
    border-width: 6px 5px 0 5px;
  }
  .triangle-after-next {
    position: relative;
  }
  .triangle-after-next::after {
    border-color: transparent;
    border-style: solid;
    content: "";
    position: absolute;
    pointer-events: none;
    top: 3rem;
    right: 2rem;
    border-left-color: #1c1c1c;
    border-width: 12px 0 12px 12px;
  }
transform / tailwindcss-truncate-multiline / css / 0: |-
  .truncate-3-lines {
    overflow: hidden;
    display: -webkit-box;
    -webkit-line-clamp: 3;
    -webkit-box-orient: vertical;
  }
  .truncate-5-lines {
    overflow: hidden;
    display: -webkit-box;
    -webkit-line-clamp: 5;
    -webkit-box-orient: vertical;
  }
